interfaces/builtin/kernel_module_load.go

```diff
@@ -0,0 +1,220 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/kmod"
+	"github.com/snapcore/snapd/snap"
+)
+
+const kernelModuleLoadSummary = `allows to statically control kernel module loading`
+
+const kernelModuleLoadBaseDeclarationPlugs = `
+  kernel-module-load:
+    allow-installation: false
+    deny-auto-connection: true
+`
+
+const kernelModuleLoadBaseDeclarationSlots = `
+  kernel-module-load:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-connection: true
+`
+
+var modulesAttrTypeError = errors.New(`kernel-module-load "modules" attribute must be a list of dictionaries`)
+
+// kernelModuleLoadInterface allows creating transient and persistent modules
+type kernelModuleLoadInterface struct {
+	commonInterface
+}
+
+type loadOption int
+
+const (
+	loadNone loadOption = iota
+	loadDenied
+	loadOnBoot
+)
+
+type ModuleInfo struct {
+	name    string
+	load    loadOption
+	options string
+}
+
+var kernelModuleNameRegexp = regexp.MustCompile(`^[a-zA-Z0-9_]+$`)
```

Comment:
I checked /lib/modules and there are some with "-" in the name, so this should be `^[-a-zA-Z0-9_]+$` I think.

Reply:
Added.